packages/duckdb-wasm/src/bindings/bindings_base.ts

```diff
@@ -134,11 +134,17 @@ export abstract class DuckDBBindingsBase implements DuckDBBindings {
 
     /** Tokenize a script */
     public tokenize(text: string): ScriptTokens {
-        const [s, d, n] = callSRet(this.mod, 'duckdb_web_tokenize', ['string'], [text]);
+        const BUF = TEXT_ENCODER.encode(text);
+        const bufferPtr = this.mod._malloc(BUF.length );
+        const bufferOfs = this.mod.HEAPU8.subarray(bufferPtr, bufferPtr + BUF.length );
+        bufferOfs.set(BUF);
+        const [s, d, n] = callSRet(this.mod, 'duckdb_web_tokenize_buffer', ['number', 'number'], [bufferPtr, BUF.length]);
         if (s !== StatusCode.SUCCESS) {
+            this.mod._free(bufferPtr);
```

Comment:
Shouldn't it be freed after reading the string? (same below in the 2 other error cases?)

Reply:
I think it's a valid question, but interface should be that duckdb_web_tokenize_buffer should only get a view of the C-array (ex-string), but are not allowed to actually reference to it and need to do a copy if query needs to be reported back in the error.

But then if this is the case, maybe code is more clear with the free just after the call. Let me double check this.

Reply:
I think I cleaned up a bit by moving the free up